lib/recursive_render.rb

```diff
@@ -0,0 +1,36 @@
+module RecursiveRender
+  #
+  # Override block_html to support parsing nested markdown blocks.
+  #
+  # @param [String] raw
+  #
+  def block_html(raw)
+    raw = unindent(raw)
+
+    if md = raw.match(/\<(.+?)\>(.*)\<(\/.+?)\>/m)
+      open_tag, content, close_tag = md.captures
+      "<#{open_tag}>\n#{recursive_render(unindent(content))}<#{close_tag}>"
+    else
+      raw
+    end
+  end
+
+  #
+  # This is jank, but Redcarpet does not provide a way to access the
+  # renderer from inside Redcarpet::Markdown. Since we know who we are, we
+  # can cheat a bit.
+  #
+  # @param [String] markdown
+  # @return [String]
+  #
+  def recursive_render(markdown)
+    Redcarpet::Markdown.new(self.class, @options).render(markdown)
+  end
+
+  def unindent(string)
+    string.gsub(/^#{string.scan(/^[[:blank:]]+/).min_by { |l| l.length }}/, "")
```

Comment:
What is this for, and is there any way we can make the code a little more easily understandable? Multiple lines and slightly more verbose code is okay. 😅 Also curious if this solution would allow, for instance

    <div>
      - list item 1
        - nested list item 1
        - nested list item 2
      - list item 2
    </div>

to work properly.